### ISSUE TITLE
UI Enhancement

### ISSUE THREAD

**Issue by wlto** [Seneca-CDOT/telescope#505]
**Background**

After taking a look at the current user interface for Telescope, I have a few suggestions about the user interface for the designers of this website and along with it are some suggestions on CSS improvements. This issue is created not to disregard the works of previous fellows but simply to suggest enhancements we can make to create a better user experience. Reading posts on a screen is by nature very tiring. Hence I think we should focus more on the user experience in regards to this aspect of our website.

**What would you like to be added**:

- **Colours**: Currently, the main colour choice of the website is mostly black, red and white. Although I understand red and white are the main colours that represent Seneca College, I would suggest **moving away from the colour red**. It creates kind of a distraction while I was scrolling through the posts. Or maybe we can use a different, perhaps a more subdued red. [This website](http://stencil.wiki/colors) has some nice colour palettes we can choose from. I hope this is useful.

- **Fonts**: Seems to be a bit inconsistent. I think we should stick to at most 2 font-families and use either sans-serif or serif for the main post content. I don't have many opinions on this.

- **Sidebar**: It seems a bit off. What I meant is that it can distract the users from the main blog content. I suggest that we can perhaps create a toggle to hide/view the sidebar. That would be great. And I'm not sure I understand the benefits of putting a Google Search bar on this menu. Perhaps someone can elaborate me on this?

- **Components**: As more work on the back-end is under construction, we should also start thinking about core UI components that are used throughout the website (buttons, links, headings, content, etc.).

- **CSS**: I believe we can all agree that this aspect of the code can be done better. I suggest using [BEM styling rules](https://blog.decaf.de/2015/06/24/why-bem-in-a-nutshell/) to help organize our CSS code for reusability. And also, it does go well with component-based libraries such as React or Angular. If possible, perhaps we can use SASS as well to get as much juice as possible from the power of CSS.

**Summary**:

- Enhanced user interface / design.
- CSS styles and structures.
  - Using BEM style guides.
  - If possible, maybe use SASS?

**Why would you like this to be added**:

As mentioned above, I believe designing blog posts that focus on user experience and interactions while maintaining their interest on the website/blog content is very important. I do realize this might not be a "present" thing. But we should discuss about this beforehand so that once it's ready for us to focus more on front-end, we will have a good starting point and guidelines to follow.

If possible, I would like to work on some if not all of these issues alongside with a couple of contributors who have had previous experience with the UI and the front-end code of this website, and the designer(s) as well.

Please let me know what you guys think!

**Comment by humphd** [comment]
Hey @wlto.  One of the things I want to do in January is to start iterating more seriously on the frontend.  We didn't have fully working backend until last week, which made it hard to build out a frontend that focused on everything we'll need going forward.

We've gotten to the point now where we've evolved a backend REST API that can be consumed in a frontend, or, by multiple frontends.  For me personally, I really want to see if I can hack the frontend out of https://github.com/GetStream/Winds and re-purpose parts of it for Telescope.  I'd also love to play with some different styles of frontends for different reading experiences (e.g., single post pages).  I'd also like to make it a PWA hosted statically, separate from the backend.

All of that to say, we can and should do a bunch of experiments with the frontend, and they don't all have to align: parallel experiments are great--that's the value of having an API.

I would encourage you to break this issue up into smaller pieces.  You've got too much in one issue, and we could tackle different aspects more easily if we separated.  For example: we can work on colour schemes separate to fonts, don't need to worry about BEM in the same bug that's doing the sidebar.  We don't need to include a discussion of SASS vs. ??? in the same issue that tries to deal with component architecture.

I want to encourage us to move toward smaller iterations vs. big change-the-world issues moving forward.
